clients/go/pkg/commands/stream_jobs_test.go

```diff
@@ -0,0 +1,198 @@
+	defer ctrl.Finish()
+
+	client := mock_pb.NewMockGatewayClient(ctrl)
+	stream := mock_pb.NewMockGateway_StreamActivatedJobsClient(ctrl)
+
+	request := &pb.StreamActivatedJobsRequest{
+		Type:    "foo",
+		Timeout: DefaultJobTimeoutInMs,
+		Worker:  DefaultJobWorkerName,
+	}
+
+	job1 := pb.ActivatedJob{
+		Key:                      123,
+		Type:                     "foo",
+		Retries:                  3,
+		Deadline:                 123123,
+		Worker:                   DefaultJobWorkerName,
+		ElementInstanceKey:       123,
+		ProcessDefinitionKey:     124,
+		BpmnProcessId:            "fooProcess",
+		ProcessInstanceKey:       1233,
+		ElementId:                "foobar",
+		ProcessDefinitionVersion: 12345,
+		CustomHeaders:            "{\"foo\": \"bar\"}",
+		Variables:                "{\"foo\": \"bar\"}",
+	}
+	job2 := pb.ActivatedJob{
+		Key:           123,
+		Type:          "foo",
+		Retries:       3,
+		Deadline:      123123,
+		Worker:        DefaultJobWorkerName,
+		CustomHeaders: "{}",
+		Variables:     "{}",
+	}
+
+	expectedJobs := []entities.Job{{ActivatedJob: &job1}, {ActivatedJob: &job2}}
+
+	gomock.InOrder(
+		stream.EXPECT().Recv().Return(&job1, nil),
+		stream.EXPECT().Recv().Return(&job2, nil),
+		stream.EXPECT().Recv().Return(nil, io.EOF),
+	)
+
+	client.EXPECT().StreamActivatedJobs(gomock.Any(), &utils.RPCTestMsg{Msg: request}).Return(stream, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), utils.DefaultTestTimeout)
+	defer cancel()
+
+	jobsChan := make(chan entities.Job, 2)
+	defer close(jobsChan)
+	err := NewStreamJobsCommand(client, func(context.Context, error) bool {
+		return false
+	}).JobType("foo").Consumer(jobsChan).Send(ctx)
+
+	assert.NoError(t, err)
+
+	// simulate job receive
+	jobs := []entities.Job{<-jobsChan, <-jobsChan}
+
+	if len(jobs) != len(expectedJobs) {
+		t.Error("Failed to receive all jobs: ", jobs, expectedJobs)
+	}
+
+	for i, job := range jobs {
+		if !reflect.DeepEqual(job, expectedJobs[i]) {
+			t.Error("Failed to receive job: ", job, expectedJobs[i])
+		}
+	}
+
+}
+
+func TestStreamJobsCommandWithTimeout(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	client := mock_pb.NewMockGatewayClient(ctrl)
+	stream := mock_pb.NewMockGateway_StreamActivatedJobsClient(ctrl)
+
+	request := &pb.StreamActivatedJobsRequest{
+		Type:    "foo",
+		Timeout: 60 * 1000,
+		Worker:  DefaultJobWorkerName,
+	}
+
+	stream.EXPECT().Recv().Return(nil, io.EOF)
+	client.EXPECT().StreamActivatedJobs(gomock.Any(), &utils.RPCTestMsg{Msg: request}).Return(stream, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), utils.DefaultTestTimeout)
+	defer cancel()
+
+	jobsChan := make(chan entities.Job, 5)
+	defer close(jobsChan)
+	err := NewStreamJobsCommand(client, func(context.Context, error) bool {
+		return false
+	}).JobType("foo").Consumer(jobsChan).Timeout(1 * time.Minute).Send(ctx)
+
+	if err != nil {
+		assert.NoError(t, err, "Failed to send request")
+	}
```

Comment:
❓ Not sure what we are asserting here in this test. Because if there is no error then nothing?

Reply:
Any calls to the mocked client which are not expected will cause the test to fail. Almost all other command tests work like this. You set up expectations with specific arguments, and only if the call matches is the test passing. The mocks in Go are very strict by default.